bittensor/dendrite.py

```diff
@@ -133,66 +133,94 @@ async def forward(
-            timeout (float, optional): The request timeout duration in seconds.
-                Defaults to 12.0 seconds.
+                The target Axons to send requests to. Can be a single Axon or a list of Axons.
+            synapse (bittensor.Synapse, optional): The Synapse object encapsulating the data. Defaults to a new bittensor.Synapse instance.
+            timeout (float, optional): Maximum duration to wait for a response from an Axon in seconds. Defaults to 12.0.
+            deserialize (bool, optional): Determines if the received response should be deserialized. Defaults to True.
+            run_async (bool, optional): If True, sends requests concurrently. Otherwise, sends requests sequentially. Defaults to True.
+            streaming (bool, optional): Indicates if the response is expected to be in streaming format. Defaults to False.
 
         Returns:
-            Union[bittensor.Synapse, List[bittensor.Synapse]]: If a single target axon is provided,
-                returns the response from that axon. If multiple target axons are provided,
-                returns a list of responses from all target axons.
+            Union[bittensor.Synapse, List[bittensor.Synapse]]: If a single Axon is targeted, returns its response.
+            If multiple Axons are targeted, returns a list of their responses.
         """
         is_list = True
         # If a single axon is provided, wrap it in a list for uniform processing
         if not isinstance(axons, list):
             is_list = False
             axons = [axons]
 
-        # This asynchronous function is used to send queries to all axons.
-        async def query_all_axons() -> List[bittensor.Synapse]:
-            # If the 'run_async' flag is not set, the code runs synchronously.
-            if not run_async:
-                # Create an empty list to hold the responses from all axons.
-                all_responses = []
-                # Loop through each axon in the 'axons' list.
-                for target_axon in axons:
-                    # The response from each axon is then appended to the 'all_responses' list.
-                    all_responses.append(
-                        await self.call(
-                            target_axon=target_axon,
-                            synapse=synapse.copy(),
-                            timeout=timeout,
-                            deserialize=deserialize,
-                        )
-                    )
-                # The function then returns a list of responses from all axons.
-                return all_responses
-            else:
-                # Here we build a list of coroutines without awaiting them.
-                coroutines = [
-                    self.call(
+        # Check if synapse is an instance of the StreamingSynapse class or if streaming flag is set.
+        streaming = (
+            issubclass(synapse.__class__, bittensor.StreamingSynapse) or streaming
```

Comment:
Should there be a warning if the two don't match? Should we expect the base class to be streaming if streaming is used?

Reply:
Yes, we should expect to subclass, but if subnet owners decide that the `bt.StreamingSynapse` is too restrictive for their purposes, they should be able to write their own class that inherits from `bt.Synapse`.

Reply:
Agree we should add a warning here.